internal/charm/charmdir.go

```diff
@@ -170,7 +170,7 @@ func (dir *CharmDir) join(parts ...string) string {
 // SetDiskRevision does the same as SetRevision but also changes
 // the revision file in the charm directory.
 func (dir *CharmDir) SetDiskRevision(revision int) error {
-	dir.SetRevision(revision)
+	dir.setRevision(revision)
```

Comment:
This method is only actually used in tests. If we want charms to be truly immutable, it should be moved off of `ChamrDir` so we aren't tempted to use it future. Perhaps a helper function in a `testing` package? What do you think?

Reply:
Correct, I don't think it's worth it right now, but we can evaluate this package to clean it up.

Reply:
I agree